### PR TITLE
Remove options, stick with builder

### PR DESCRIPTION
**Pull request by reyang**
Related to #1858 and #1833.

## Changes

I would propose that we only use **builder as the configuration**, instead of having both builder and options pattern.
This could also help us to have a consistent model when we convert raw metrics to logs, activities to sampled metrics, activity events to logs, and logs to activity events - as discussed [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1833#issuecomment-782560618).

Couple benefits:
* Smaller API surface
* Avoid the confusion of having two ways of doing things
* Builder pattern allows people enable/disable things, rather than configuration - once set you're done and there is no way for the others to disable it in their code.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Design discussion issue #1853
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by ejsmith** [comment]
I really don't get the motivation for this. This is a pretty standard pattern. Options is the root thing that can just be new'd up and you can configure the options yourself. But a builder wrapper around that to provide a fluent API for setting up your configuration. It's the exact same pattern that appears about 1000 times all over the .NET generic host and ASP.NET and many many other systems.

**Comment by alexvaluyskiy** [comment]
OpenTelemetry docs says that SDK should not have any breaking changes. At least for one year. I think, this PR changes the way to configure of OpenTelemetry, and break the compatibility.

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,8 @@ please check the latest changes
 ## Unreleased
 
 * Added `TracerProviderOptions` and `SetErrorStatusOnException`.
```
Comment (cijothomas) [inline src/OpenTelemetry/CHANGELOG.md:12]: `TracerProviderOptions` can be removed from changelog as well

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
> OpenTelemetry docs says that SDK should not have any breaking changes. At least for one year. I think, this PR changes the way to configure of OpenTelemetry, and break the compatibility.

There is no breaking change. This PR is reverting an option which was introduced last week (and never released, as indicated by changes to .publicapi.unshipped.txt).

File: src/OpenTelemetry/CHANGELOG.md (hunk 1 above)
Comment (reyang) [reply src/OpenTelemetry/CHANGELOG.md]: Updated.

**Review by alanwest — APPROVED** [review]

File: src/OpenTelemetry/Trace/TracerProviderBuilderExtensions.cs [hunk 2]
```diff
@@ -22,6 +22,23 @@ namespace OpenTelemetry.Trace
...
     public static class TracerProviderBuilderExtensions
     {
+        /// <summary>
+        /// Sets whether the status of <see cref="System.Diagnostics.Activity"/>
+        /// should be set to <c>Status.Error</c> when it ended abnormally due to an unhandled exception.
+        /// </summary>
+        /// <param name="tracerProviderBuilder">TracerProviderBuilder instance.</param>
+        /// <param name="enabled">Enabled or not. Default value is <c>true</c>.</param>
+        /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
+        public static TracerProviderBuilder SetErrorStatusOnException(this TracerProviderBuilder tracerProviderBuilder, bool enabled = true)
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Trace/TracerProviderBuilderExtensions.cs:32]: @reyang Just curious, what's the use case you were thinking of for adding with `enabled = false`?

**Pull request merged**